### PR TITLE
JA.13 and JA.14

### PR DESCRIPTION
**Pull request by gartmand**
* Added the validation tests for our domain models. 

* Added the validation to pass the validation tests. It now runs when an Employee, Skill, or Category create/update endpoint is used.

* Formatting and cleanup

### CODE REVIEW

File: src/main/java/com/excella/reactor/domain/Address.java [hunk 1]
```diff
@@ -11,8 +12,9 @@
   private String line2;
   private String city;
 
+  @NotNull
   @Size(min = 2, max = 2)
```
Comment (vickumar1981) [inline src/main/java/com/excella/reactor/domain/Address.java]: if the rule is 2 letters for the state, then why not use `@Pattern` and supply the regex instead?

File: src/main/java/com/excella/reactor/domain/Bio.java [hunk 2]
```diff
@@ -10,25 +11,31 @@
 import javax.persistence.Embeddable;
 import javax.persistence.EnumType;
 import javax.persistence.Enumerated;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
 import lombok.Data;
 
 @Data
```
Comment (vickumar1981) [inline src/main/java/com/excella/reactor/domain/Bio.java]: does this class have to be `@Data` or can it be changed to `@Value`?  I wasn't sure if there was some issue with the ORM where it has to be able to set the `Id` field.

File: src/main/java/com/excella/reactor/domain/Address.java (hunk 1 above)
Comment (vickumar1981) [inline src/main/java/com/excella/reactor/domain/Address.java]: if the rule is 2 letters for the state, then why not use @Pattern and supply the regex instead?

Comment (gartmand) [reply src/main/java/com/excella/reactor/domain/Address.java]: The rule from the ticket is just that it's of length 2. I think we should evaluate this along with other specs (such as that bio first and last name are just required with no other spec). Would need to talk to sync up with tcp-angular on the validation spec again. @kentherussell Was JA.13 based on previous talks with the angular team?

Comment (kentherussell) [reply src/main/java/com/excella/reactor/domain/Address.java]: It's based on the model that was given, in addition to previous talks.

File: src/main/java/com/excella/reactor/domain/Bio.java (hunk 2 above)
Comment (gartmand) [reply src/main/java/com/excella/reactor/domain/Bio.java]: Adding a ticket to refactor to use value classes where appropriate.

File: src/main/java/com/excella/reactor/domain/Address.java (hunk 1 above)
Comment (gartmand) [reply src/main/java/com/excella/reactor/domain/Address.java]: Added a comment to JA.12 to finalize these when discussing the data contract.

**Pull request merged**